### PR TITLE
:seedling: Emit a webhook warning when OvfEnv or ExtraConfig transport is specified

### PR DESCRIPTION
**Pull request by aruneshpa**
**What does this PR do, and why is it needed?**
This change emits a warning when a user specifies OvfEnv transport in a VM's spec.  In order to support warnings, modify the BuildValidationResponse so it can now take warnings from any validations.
At some point, we should move our webhooks to controller-runtime generated webhooks which will remove the need to synthesize the response from errors and warnings ourselves.

**Which issue(s) is/are addressed by this PR?** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
None

**Please add a release note if necessary**:
```release-note
Emit a webhook warning when OvfEnv or ExtraConfig transport is specified
```

### CODE REVIEW

File: webhooks/virtualmachine/v1alpha1/validation/virtualmachine_validator_unit_test.go
Comment (bryanv) [inline webhooks/virtualmachine/v1alpha1/validation/virtualmachine_validator_unit_test.go:429]: From the v1a2 work there is a pending item to start to move away from the table driven tests here because they have become IMO untenable and unreadable for a lot of stuff (there are a lot more network and bootstrap stuff in v1a2 that I have no desire to express with bools).

How about for this MR we leave the table stuff as-is (and add an assert in validateCreate that warning is always empty) to keep this diff smaller & focused, and lay down the start of individual Context/It?

**Comment by bryanv** [comment]
Can we also add an integration test the warning parts of the response? We just need one to validate it going forward - there are a lot of duplicate unit & integration tests there that otherwise don't add much value.

There should be an item in the BL to move BuildValidationResponse() "up" one caller to reduce some duplication but interia has made it difficult to really prioritize that.

Like I showed you yesterday, with v1a2 conversion we now have the pieces in place to use the ctrl-runtime webhooks but it is a fair amount of work - going the CustomValidator route is the prb least miserable way - and we don't get a configured as-we expect k8s client plumb down from above (but we really need to move towards the webhooks dealing with just internal consistency and defaulting instead of sometimes needed external objects). I had briefly considered converting some of the "easy" type webhooks as part of the v1a2 work but dropped it due to the churn involved. I'd prefer we wait until v1a2 has landed, and the dust has settled.

File: webhooks/virtualmachine/v1alpha1/validation/virtualmachine_validator.go
Comment (dilyar85) [inline webhooks/virtualmachine/v1alpha1/validation/virtualmachine_validator.go:216]: Should we warn `ExtraConfig` here as well since it's deprecated according to our docs at https://vm-operator.readthedocs.io/en/latest/concepts/workloads/guest/#extraconfig?

File: webhooks/virtualmachine/v1alpha1/validation/virtualmachine_validator.go
Comment (zyiyi11) [inline webhooks/virtualmachine/v1alpha1/validation/virtualmachine_validator.go:68]: `vAppConfig` see: https://github.com/vmware-tanzu/vm-operator/blob/main/api/v1alpha1/virtualmachine_types.go#L192

**Review by bryanv — APPROVED** [review]

**Review by dilyar85 — APPROVED** [review]
LGTM.

**Pull request merged**
